pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -156,6 +157,7 @@ func (s *Service) CreateASG(scope *scope.MachinePoolScope) (*expinfrav1.AutoScal
 		DefaultCoolDown:      scope.AWSMachinePool.Spec.DefaultCoolDown,
 		CapacityRebalance:    scope.AWSMachinePool.Spec.CapacityRebalance,
 		MixedInstancesPolicy: scope.AWSMachinePool.Spec.MixedInstancesPolicy,
+		AvailabilityZones:    scope.AWSMachinePool.Spec.AvailabilityZones,
```

Comment:
I think we need extra logic around this as the availability zones are available on the `MachinePool` via the failure domains.

Perhaps we should add a `AvailabilityZones` function on the [MachinePoolScope](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/65267e30239ff31097029d137e62b64e82b3473b/pkg/cloud/scope/machinepool.go) that:
1. Returns availability zones explicitly declared on the AWSMachinePool spec
2. If the spec doesn't have azs defined then use the non-controlplane failure domians on the `MachinePool`

Reply:
Good point! I have added the suggested function.